src/pilot/assign_role.py

```diff
@@ -1160,11 +1168,49 @@ def select_os_volume(os_volume_size_gb, ironic_client, drac_client, node_uuid):
 
                     break
 
+            # Note: This code block represents single disk scenario.
             if raid_size_gb == 0:
-                raise RuntimeError("There must be either a virtual disk that "
-                                   "is not a RAID 0 to install the OS on, or "
-                                   "os-volume-size-gb must be specified")
+                if virtual_disk_docs:
+                    raid_disk_ids = []
+                    for virtual_disk_doc in virtual_disk_docs:
+                        fqdd = get_fqdd(virtual_disk_doc, DCIM_VirtualDiskView)
+                        raid_type = utils.find_xml(
+                            virtual_disk_doc,
+                            'RAIDTypes',
+                            DCIM_VirtualDiskView).text
+
+                        if raid_type == RAID0:
+                            raid_size = get_size_in_bytes(virtual_disk_doc,
+                                                          DCIM_VirtualDiskView)
+                            raid_size_gb = int(raid_size) / units.Gi
+                            raid_disk_ids.append(raid_size_gb)
+                    if not len(raid_disk_ids) == 1:
+                        raise RuntimeError(
+                            "There must be a non-RAID0 virtual disk,"
+                            "a single disk RAID0, or a single JBOD disk"
+                            "to install the OS on,"
+                            "or os-volume-size-gb must be specified.")
+                    os_volume_size_gb = raid_disk_ids[0]
+                else:
+                    physical_disk_view_doc = drac_client.enumerate(
+                                                 DCIM_PhysicalDiskView)
+                    physical_disk_docs = utils.find_xml(
+                        physical_disk_view_doc,
+                        'DCIM_PhysicalDiskView',
+                        DCIM_PhysicalDiskView,
+                        True)
+                    disks = [get_size_in_bytes(physical_disk_doc,
+                                               DCIM_PhysicalDiskView)
+                             for physical_disk_doc in physical_disk_docs]
+                    if not len(disks) == 1:
+
+                        raise RuntimeError(
+                            "There must be a non-RAID0 virtual disk,"
+                            "a single disk RAID0, or a single JBOD disk"
+                            "to install the OS on,"
+                            "or os-volume-size-gb must be specified.")
 
+                    os_volume_size_gb = int(disks[0]) / units.Gi
```

Comment:
Does the check below to see if there are any physical disks that do not back the RAID that are the same size as the RAID need to be accommodated by this new functionality? I do not see `raid_physical_disk_ids` being set.

Reply:
i will add physical disk  which is equal to `RAID0` type to `raid_physical_disk_ids` list